### ISSUE TITLE
MONEY BAR?

### ISSUE THREAD

**Issue by forran** [ValveSoftware/halflife#702]
It would be nice to have player moneys shown in TAB as it is in CS:GO.

**Comment by mikela-valve** [comment]
Hi all, I've added this in the latest CS beta. You only see the money for members of your current team and only when you are a T or CT (i.e. spectators see nothing).

![146CBBB1-4A95-4FFC-9858-F1D708948EF4](https://user-images.githubusercontent.com/49570952/56451871-ba587100-62e2-11e9-9603-27f742ba5a1a.jpg)

![AF5207FC-165C-4D34-9C03-DD345B21EA7E](https://user-images.githubusercontent.com/49570952/56451880-be848e80-62e2-11e9-83ec-5227321dcb63.jpg)

**Comment by RauliTop** [comment]
@mikela-valve 
It's great, but it should have a server cvar to disable this feature.

For example, in some zombie servers or other mods that doesn't use money.

**Comment by Mistrick** [comment]
@mikela-valve would be nice if amxx plugin(or metamod) can set own name for this colomn and set format.
Use like assist or modney/assiist with format `$1000/4`

**Comment by d3m37r4** [comment]
It would be great if reduction was used for long nicknames in scoretab.

**Comment by wopox1337** [comment]
In last HLDS update be added new CBasePlayer members such as `m_iLastAccount & m_tmNextAccountUpdate`

![image](https://user-images.githubusercontent.com/18553678/56588354-d9f5df00-65eb-11e9-8686-fd69e1e72c93.png)
It's good. But why not at the end of the list?

@mikela-valve please move this members to end of list for better back-compability.

**Comment by mikela-valve** [comment]
Thanks @wopox1337, that was something I'd intended to change before committing but missed.

**Comment by wopox1337** [comment]
@mikela-valve also can you remove 16000 cap from client-side ScoreBoard `Account` value and make Money column a little bit widder
```
name       245x10 -> 225x10
class         56x10
account   25x10 -> 45x10
frags         40x10
deaths     46x10
ping          46x10
```

![image](https://user-images.githubusercontent.com/18553678/56679326-793ed300-66cd-11e9-85ec-bf80ed5a2ee1.png)

**Comment by mikela-valve** [comment]
Can do, I only set the limit of 16000 to prevent it from flowing outside of the column for larger values, though I didn't test in low quality mode so I didn't realize it was already being truncated there.

**Comment by mikela-valve** [comment]
A commenter on another issue also mentioned that showing teammate healths might be even more helpful than money values. I'd be happy to switch it to health if that would be better or add another column for teammate health.

**Comment by perforatorRU** [comment]
@mikela-valve, maybe like this? This is more informative for players.
![1920x1080](https://user-images.githubusercontent.com/12104741/56691356-68e82180-66e8-11e9-8996-1108297f523e.jpg)

**Comment by GiovaniFerraroTrivelli** [comment]
I don't know how scoreboard works, but as far as I know, there is no possibility to modify scoreboard using AMXX plugins. Is there a way that columns were added dynamically using server cvars, or better, plugins could add custom columns?

In some mods, like [TTT mod](https://forums.alliedmods.net/showthread.php?t=238780), for example, if you see Health of teammates changing is an advantage. Money is not problem because you can set $0 on every moment. Others mods, like [Soccerjam](https://forums.alliedmods.net/showthread.php?t=41447) will be amazing if we could add column for goals, for assists, appart from kills, etc.

If it's not possible, with Money and/or Defuse Kit like @perforatorRU said are great additions.

**Comment by mikela-valve** [comment]
@perforatorRU That could all be done I think. The one concern I have is that adding too many columns or too wide of columns will eat into the space available for player names, although I presume people would rather see more information than be able to see full player names.

@GiovaniFerraroTrivelli Dynamically switching out the meaning of a column (that is, changing the name, position, size and what data is displayed) is not really a trivial thing in the scoreboard code so I'm not sure how accommodating the scoreboard can be to mods. I can make it possible to suppress columns from being shown based on a cvar so unnecessary columns are hidden at least.

**Comment by basuritashka** [comment]
Being able to add/delete columns via cvar seems a good idea :)

**Comment by d3m37r4** [comment]
@mikela-valve, I suggest adding a shorthand for long player names. At moment they are just cropped.

**Comment by shelru** [comment]
Make a cvar to disable all this features in tab.

**Comment by shelru** [comment]
Maybe Money column must be there? Add some cvar like mp_showplayersmoney 0/1/2, 0 - disabled, 1 - only T, 2 - only ct, 3 - each team can see. Make it more customizable.
![image](https://user-images.githubusercontent.com/9306996/56748478-fe3ef080-6790-11e9-83c8-c15d8342a87f.png)

**Comment by hashnimmo** [comment]
> Maybe Money column must be there?
> ![image](https://user-images.githubusercontent.com/9306996/56748478-fe3ef080-6790-11e9-83c8-c15d8342a87f.png)

I agree with this. It can't be considered as a column until it's placed in the yellow column row above. The place where @mikela-valve  has currently placed the red "Money" text should have been be used to show the total team money instead of considering it as a column.

**Comment by mikela-valve** [comment]
Updated in 'Exe build: 11:12:36 May 21 2019 (8244)'.

There are now columns for both health and money that can be enabled/disabled on the client. There are also server cvar's for controlling which teams should have their health/money values sent to other clients. The column headers now show at the top like they should and I additionally fixed the column headers incorrectly being placed farther to the right than they should be when avatars are enabled. I also added support for showing when CT players have a defusal kit ('Defusal Kit' in the 'Bomb'/'VIP'/'Dead' section).

The new cvar's for the client are:
`scoreboard_showmoney` 
`scoreboard_showhealth`
Both can be set to `0` to disable their column and `1` to enable their column.
Both are set to `1` by default.

The new cvar's for the server are:
`mp_scoreboard_showmoney`
`mp_scoreboard_showhealth`
There are four values these can be set to:
* `0` don't send any update for this field to any clients
* `1` send only Terrorist money/health fields to all clients
* `2` send only CT money/health fields to all clients
* `3` send Terrorist and CT money/health fields only to members of the same team
* `4` send Terrorist and CT money/health fields to all clients
These default to `3` (sending money and health updates to teammates only).

**Comment by PartialCloning** [comment]
When you disable the cvar client side the column is completely removed. Can this be replicated for the server side cvar too? Remove the column if the server has it disabled? There is too much wasted space and the name font size is still reduced even when the server side cvar is disabled.

**Comment by PredatorFly** [comment]
![Uploading fix.jpg…]()

Spanish
Health --> Salud
Money --> Dinero

**Comment by 10th** [comment]
If you disable the column visible shift right to make room for long nicknames.

More compact solutions:
Heals --> HP% (currently the server name neglective "Heals").
100% --> 100 (1 character for nick is better if not added to the "Dead"column).
0% --> Dead (Defusal Kit, Bomb and VIP to leave in another column).

**Comment by GBCS-PRO** [comment]
I'd suggest adding the same feature for buymenu, just like in CZ.

**Comment by RauliTop** [comment]
> If the column is disabled, it is better to shift to the right visible columns to free up space for long nicknames.
> 
> More compact solutions:
> Heals --> HP% (currently "Heals" overlap on the server name 600х480).
> 100% --> 100 (1 character for nick is better if not added to the column "Dead").
> 0% --> Dead (Defusal Kit, Bomb and VIP to leave in another column).

If columns are disabled it's better to have more space to the nickname or other colums, YES.

Health will be a good option to be replaced to HP. To avoid translations issues about lenght.
Delete the '%' also it's a good option. Because on mods you can have more than 100 health or you just use that column to show another feature. So, the use of percentage seems useless.

Be careful about replace 0% to DEAD. Because the cvars can be disabled:
```
scoreboard_showhealth (on client)
mp_scoreboard_showhealth (on server)
```

So, it will be prudent to don't change it. Remain DEAD in the same position.
We don't want to create more tangle

**Comment by Splatt581** [comment]
I noticed that server sends `HealthInfo` and `Account` usermessages in each packet, sometimes even several:

```
 c <-- sz=101 seq=3531 ack=18344 rel=1 tm=139.796341
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
Msg: svc_time
Msg: svc_clientdata
Msg: svc_deltapacketentities
 c <-- sz=101 seq=3532 ack=18350 rel=1 tm=139.829956
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
Msg: svc_time
Msg: svc_clientdata
Msg: svc_deltapacketentities
 c <-- sz=89 seq=3533 ack=18355 rel=1 tm=139.861938
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
UserMsg: HealthInfo (size 5)
UserMsg: Account (size 5)
Msg: svc_time
Msg: svc_clientdata
Msg: svc_deltapacketentities
```

This is despite the fact that I am the only player on the server and my health and money do not change. This significantly increases the game traffic. Is it possible to make that these messages are sent only when there is a direct change in the amount of money and health, or at least once a second or so?

**Comment by mikela-valve** [comment]
Thanks @Splatt581, those messages are only supposed to be sent every five seconds or when the underlying stats are changed. It looks like I wasn't updating one of the stat-tracking variables so it was sending the message every update. I'll have that fixed shortly.

**Comment by mikela-valve** [comment]
@Splatt581 The erroneous high frequency updates should be fixed as of beta  'Exe build: 16:32:50 May 22 2019 (8245)'.

**Comment by mikela-valve** [comment]
The latest beta changes some of the field sizes to hopefully fit things better. The name column has been lengthened a bit even when the health and money columns are enabled. Class (dead/vip/bomb) has been reduced back to where it was before I added Defusal Kit which I've now shortened to D. Kit. Based on one of the screenshots focused I've lengthened the kills column a bit to fit translated strings. I've added a small fixed-length spacer column in between the name and class columns that will keep the name from partially overwriting the class column when it gets too long.

The health and money columns now will only show up if the server is sending updates for those fields, showing only one or the other if the server is not sending both. Disabling and enabling the scoreboard columns via the client cvar's now takes effect immediately rather than the next time the scoreboard is initialized (map change, server connect, etc).

**Comment by oaus** [comment]
] version
Protocol version 48
Exe version 1.1.2.7/Stdio (cstrike)
Exe build: 16:32:50 May 22 2019 (8245)

] mp_scoreboard_showmoney
Unknown command: mp_scoreboard_showmoney
] mp_scoreboard_showhealth
Unknown command: mp_scoreboard_showhealth
] scoreboard_showhealth
Unknown command: scoreboard_showhealth
] scoreboard_showmoney
Unknown command: scoreboard_showmoney

**Comment by mikela-valve** [comment]
@oaus Is it possible that you're opted into the HL1 beta but not the CS beta? If your CS binaries are somehow still on the public version those cvar's wouldn't exist as they're created by the CS client.

**Comment by oaus** [comment]
> @oaus Is it possible that you're opted into the HL1 beta but not the CS beta? If your CS binaries are somehow still on the public version those cvar's wouldn't exist as they're created by the CS client.

Ok, that was a strange thing, I was on the BETA version on Steam and for some strange reason I stopped being in it.

mp_scoreboard_showmoney 1
mp_scoreboard_showhealth 1
![image](https://user-images.githubusercontent.com/43151226/58219379-dd05ed00-7cc7-11e9-9080-2d810b3b568a.png)

and:
mp_scoreboard_showmoney 0
mp_scoreboard_showhealth 0
![image](https://user-images.githubusercontent.com/43151226/58219411-fc047f00-7cc7-11e9-9def-96440f165337.png)


Is it possible to remove the "Health and money" fields if the value of "mp_scoreboard_showmoney" and "mp_scoreboard_showhealth" are 0?

I know that the fields are deleted if the client has its values ​​at 0, but it is possible to eliminate them by the server.

Also the field on the left side of the "score" overlap a bit with the translation of "score" in Spanish.

**Comment by mikela-valve** [comment]
Actually if you enter the server with either of the `mp_scoreboard` variables set to 0 you should not see whichever of the fields that are disabled in the client. 

The server doesn't have a regular way to sync up cvar's with the client so any information like the value of the server cvar's has to be pushed to the client, so what's happened here is the client saw health and money updates sent from the server so it enabled the columns but then the server stopped sending the updates so the columns are there because the client is expected the server to send more updates. I don't think this should happen on a running server very often since I don't think there is a reason a server would want to shut the updates off mid-map (the columns reset on map and server changes).

The problem with the score column is that it's not sized correctly for the Spanish translations, that is the column width is too small so the money column is being placed too close to it. I did try increasing the width of that column some for this update but it's apparently not enough (if you compare it to your previous screenshot the money column doesn't overlap as much).

**Comment by GiovaniFerraroTrivelli** [comment]
> The problem with the score column is that it's not sized correctly for the Spanish translations, that is the column width is too small so the money column is being placed too close to it. I did try increasing the width of that column some for this update but it's apparently not enough (if you compare it to your previous screenshot the money column doesn't overlap as much). I'll try either increasing the width of that column a bit more in a future release or setting the health and money columns to be left aligned instead of right aligned.

Try changing "Puntuación" to "Puntos" for "Score" translation in Spanish. I think it is a simple solution but it won't solve the problem if other languages have a long string too.

**Comment by RauliTop** [comment]
I said how to solve length problem at the scoreboard for all languages. But noboby cared...
> * Health will be a good option to be replaced to HP. To avoid translations issues about lenght.
> * Money will be a good option to be replaced to $. To avoid translations issues about lenght.
> * Delete the '%' and the '$' also it's a good option. Because on mods you can have more than 100 health or you just use that column to show another feature. So, the use of percentage or dollar symbol seems useless.

**Comment by yek-ta** [comment]
If it will be like CS:GO score board, I think you can add "Changing Team Name" too. PUG players will like it.
And amxmodx authors must change "Defusal Kit/C4/Dead" texts, like server name. This will please everyone.
![56451871-ba587100-62e2-11e9-9603-27f742ba5a1a](https://user-images.githubusercontent.com/37214615/58248345-8a2e4300-7d64-11e9-8776-bbac2ca0822c.jpg)

**Comment by afwn90cj93201nixr2e1re** [comment]
omg

**Comment by afwn90cj93201nixr2e1re** [comment]
Is there team C, D, E, F, G? Why we can't just hold it as it is.

**Comment by yek-ta** [comment]
This is just a suggestion. 1.6 players are running out. At least they'll get their attention. If there is no player, these updates will not have a purpose. This game must be passed from generation to generation. I think this happens like this.

**Comment by afwn90cj93201nixr2e1re** [comment]
Just remove new column's. This is stupid way to allow server customize client. Create cvar cl_show_money/cl_show_health, client must have option to show/hide, that's mustn't be forwarded from server.

**Comment by PartialCloning** [comment]
The server has the option to enable/disable columns as of the last beta update.

The '$' sign should remain. 5 different integer based rows next to each other without a single visual cue is an eyesore. It's consistent between the scoreboard and HUD. The '%' is unnecessary as the HUD does not use '%' to represent health.

**Comment by RauliTop** [comment]
> The server has the option to enable/disable columns as of the last beta update.
> 
> The '$' sign should remain. 5 different integer based rows next to each other without a single visual cue is an eyesore. It's consistent between the scoreboard and HUD. The '%' is unnecessary as the HUD does not use '%' to represent health.

Well, I thought the same when I wrote it.
But having the '$' will make column not customizable using plugins...

**Comment by afwn90cj93201nixr2e1re** [comment]
> 
The new cvar's for the client are:
scoreboard_showmoney
scoreboard_showhealth
Both can be set to 0 to disable their column and 1 to enable their column.
Both are set to 1 by default.

**Comment by Shalun22** [comment]
You plan to add support for this option in the CS:CZ ?

**Comment by 10th** [comment]
Now if mp_scoreboard... is 0, then columns are hidden.
Is it possible to hide the columns if the server is old and has no new variables?

**Comment by afwn90cj93201nixr2e1re** [comment]
How you gonna check it?
Only one way is - hide column until money message come.

**Comment by BlackShadow306** [comment]
@mikela-valve instead of writing Defusal Kit wouldn't be nice if we put it as sprite? With like that it would take less space in scoreboard.

I tried to replicate this with photoshop:

https://i.imgur.com/GYEPxw4.jpg

**Comment by mikela-valve** [comment]
@BlackShadow306 I'd rather keep it in the class text area if possible as a CT analog for Bomb, but I'll consider it for the future if any other changes are made to those scoreboard entries. 

I did shorten it from 'Defusal Kit' to 'D. Kit' in the latest release so it doesn't take up much more room than VIP/Bomb/Dead and should fit into the space currently allocated for those (as the space is a bit larger than needed for four characters).

**Comment by Splatt581** [comment]
Dead players can see health of enemies.

As I understand it, a client can get health of other players in scoreboard in two ways - when received user messages `HealthInfo` or `StatusValue`. Sending `StatusValue` is controlled by cvar `mp_playerid`. The bug is that regardless of value of cvar `mp_playerid`, server sends health of live players to dead players and spectators / unassigned if they look at them. More details can be seen on the video:
[YouTube link](https://www.youtube.com/watch?v=5LTAxlIz_c4)

I also propose for cvars `mp_scoreboard_showhealth` / `mp_scoreboard_showmoney` to enter a new value `5`, at which server will send health and money to teammates and spectators. This will be needed to correctly HLTV monitoring for a competitive game.

**Comment by afwn90cj93201nixr2e1re** [comment]
@mikela-valve check it.

**Comment by mikela-valve** [comment]
Thanks @Splatt581, I've fixed the issue with the `StatusValue` messages setting the same value that the scoreboard uses. I've also added the value for the scoreboard cvar's that you mentioned. This will be included in the next beta update.

**Comment by mikela-valve** [comment]
This is updated in build 8254 with the fixes referenced above.

**Comment by oaus** [comment]
"dead" should not appear in the place where life was?
![image](https://user-images.githubusercontent.com/43151226/58758747-1f57d700-84dd-11e9-95e9-0f19f20a6b16.png)

also, the money text overlays a bit with "Puntuación"
![image](https://user-images.githubusercontent.com/43151226/58758744-123ae800-84dd-11e9-8e52-4c3579281f73.png)

**Comment by mikela-valve** [comment]
@oaus Dead is appearing where VIP/Bomb/D. Kit does, not where HP is. Health values are right justified so they appears directly below the right side of HP, not in the middle of the screen like VIP does.

Puntación is still overlapping due to it being too long of a header for the score field, so it will overlap Money until I either change the justification of the Money header or change to Puntos as you recommended (which is likely what I'll do since it will fit better at 640x480 and 800x600).

**Comment by oaus** [comment]
Puntaje (640x480) (Puntos was recommended by: @GiovaniFerraroTrivelli )
![image](https://user-images.githubusercontent.com/43151226/58759075-c179be00-84e1-11e9-84a8-545f2e71203b.png)

**Comment by mikela-valve** [comment]
Sorry, I hadn’t looked back to see who had recommended that so I thought you had. So you’d recommend Puntaje as a closer translation for this?

**Comment by afwn90cj93201nixr2e1re** [comment]
Maybe you can create an issue where we can give you some transl. recommendation's?

**Comment by d3m37r4** [comment]
![58759075-c179be00-84e1-11e9-84a8-545f2e71203b](https://user-images.githubusercontent.com/26052038/58759603-dc3a4a00-8556-11e9-9160-ad8905a3c100.png)
Perhaps should use the space on the right, according to what is marked in green? Make alignment columns (ping, kills and frags).

**Comment by afwn90cj93201nixr2e1re** [comment]
В таком случае на больших разрешениях всё 'поплывёт'.

**Comment by GiovaniFerraroTrivelli** [comment]
> Sorry, I hadn’t looked back to see who had recommended that so I thought you had. So you’d recommend Puntaje as a closer translation for this?

@mikela-valve Puntos and Puntaje have the same meaning in this context. Puntaje is the sum of all "puntos" (score) from a player. So choose what you think it's better for spacing 👍

**Comment by RauliTop** [comment]
> @mikela-valve Puntos and Puntaje have the same meaning in this context. Puntaje is the sum of all "puntos" (score) from a player. So choose what you think is better for spacing 👍

Who uses Puntaje?

Puntos it's better

**Comment by eat1k** [comment]
I agree. “Puntos” is better than “Puntaje”.

**Comment by mikela-valve** [comment]
Great, I'll update the Spanish localization with that change.

**Comment by mikela-valve** [comment]
@afwn90cj93201nixr2e1re There is a localization issue created that you can add suggestions to, #2311 .

**Comment by mikela-valve** [comment]
@d3m37r4 That space is currently only there because the resolutions with height<768 don't display the avatar column. There is an open request to enable that for all resolutions so I thought I'd leave that space until I determine if that will work well or not.

**Comment by mikela-valve** [comment]
@basuritashka @GiovaniFerraroTrivelli The new scoreboard headers should have proper Spanish localizations in CS beta 8260. The default MOTD of the day should be updated as well.

**Comment by basuritashka** [comment]
@mikela-valve We have been talking about that a few mins ago, we can just replace `Salud` with `Vida` if there is a problem with that.

**Comment by CS-PRO1** [comment]
I just launched CS beta 8260 .. And I don't see the D.kit and Health in scoreboard anymore.. They were reverted for testing something else or what exactly?

**Comment by eat1k** [comment]
> I just launched CS beta 8260 .. And I don't see the D.kit and Health in scoreboard anymore.. They were reverted for testing something else or what exactly?

`scoreboard_showhealth` 1
`scoreboard_showmoney` 1

**Comment by CS-PRO1** [comment]
> > I just launched CS beta 8260 .. And I don't see the D.kit and Health in scoreboard anymore.. They were reverted for testing something else or what exactly?
> 
> `scoreboard_showhealth 1`
> `scoreboard_showmoney 1`

I have them set by default and still not showing. I tried 2 3 4 still same.

**Comment by mikela-valve** [comment]
@CS-PRO1 Those are only 0 or 1 to disable/enable on the client. `mp_scoreboard_showhealth` and `mp_scoreboard_showmoney` on the server are what have values 0-5. Check that the server cvar’s are set to show values for the team that you’re on (4 to be safe) as I recently changed the client to not show the health or money columns if the server is not sending values for any of the players so they don’t show up on servers where it’s disabled.

**Comment by CS-PRO1** [comment]
> @CS-PRO1 Those are only 0 or 1 to disable/enable on the client. `mp_scoreboard_showhealth` and `mp_scoreboard_showmoney` on the server are what have values 0-5. Check that the server cvar’s are set to show values for the team that you’re on (4 to be safe) as I recently changed the client to not show the health or money columns if the server is not sending values for any of the players so they don’t show up on servers where it’s disabled.

@mikela-valve I was testing it on offline local server idk..

**Comment by CS-PRO1** [comment]
Oh and btw, I think options for this feature should be added to Game settings in "Create server"

**Comment by eat1k** [comment]
It works fine for me in the latest beta version: Exe build: 10:20:22 Jun  6 2019 (8260)
![image](https://user-images.githubusercontent.com/31934972/59066396-40d70b00-88af-11e9-9452-4eec2893e530.png)

**Comment by mikela-valve** [comment]
@CS-PRO1 Those variables still need to be set properly on a listen server to see the columns, they're just set through the client rather than a DS console. If you have them set to `4` and the client cvar's set to `1` and still don't see the columns then there might be something else going awry. I'd try switching out and back into the beta if so to see if that fixes it.

**Comment by CS-PRO1** [comment]
![score](https://user-images.githubusercontent.com/50755959/59067100-6bc65c80-88b9-11e9-8201-f5df764c637e.png)
That's mine currently. Btw I just click new game and pick random map not that big deal :d
I tried setting cvar before starting game and after. Nothing changes..

**Comment by mikela-valve** [comment]
If you set `mp_scoreboard_showhealth` and `mp_scoreboard_showmoney` to `4` after the map is started you still never see those columns show up? You won't be able to set them before starting the server for the first time as the cvar's aren't registered yet. 

Try running with `-dev -condebug` and post anything weird that might show up in the console.

**Comment by BlackShadow306** [comment]
![20190607002804_1](https://user-images.githubusercontent.com/30868395/59067801-315dbf00-88bb-11e9-8de8-3f5538b1302f.jpg)

@CS-PRO1 I think it's on your side. I haven't changed any cvar or anything and i can see all columns. Try to do fresh install if it's possible.

**Comment by CS-PRO1** [comment]
I found the problem's source, it seems that installing this bot addon breaks the function:
https://steamcommunity.com/app/10/guides/?searchText=add+bot&browsefilter=trend&requiredtags%5B%5D=-1#scrollTop=0

I found out by letting steam verify CS directory and it replaced 5-6 files. Sorry for disturbing you for such a bug but yea I should've known that the bot addon doesn't support new features 'n stuff. I'll uninstall it entirely.

**Comment by afwn90cj93201nixr2e1re** [comment]
On low res. same problem.
Maybe you could create some switch for .res files?
On hl load check, which one width using right now, and select them.
Coz that's not good:
![image](https://user-images.githubusercontent.com/9306996/59070194-e51a7b00-88ca-11e9-82b8-09c9498364a1.png)
Same with eng localization:
![image](https://user-images.githubusercontent.com/9306996/59070302-51957a00-88cb-11e9-9207-d2f5e888ddf0.png)

![image](https://user-images.githubusercontent.com/9306996/59070264-23179f00-88cb-11e9-8e58-09bdd89d879f.png)
It's from valve_russian.txt

Also, can you fix closes icon margin?
![image](https://user-images.githubusercontent.com/9306996/59070330-6d991b80-88cb-11e9-88b0-6e7012dc6c8e.png)
Seem's like it's should be centered. Maybe it placed in GameUI.dll, but i haven't .so file to reverse it.

**Comment by GiovaniFerraroTrivelli** [comment]
@mikela-valve Like @basuritashka said, please change "Salud" to "Vida" if it fits better. If neither of them fit, it wouldn't be bad if you keep it as "HP" like English default text.

@basuritashka En todas partes "Health" es traducido como "Salud" (lo cuál es su traducción más acertada, pero no es común escucharlo como tal dentro del juego). ¿Ves necesario modificarlo a "Vida" para que quede en todos los casos igual?

**Comment by basuritashka** [comment]
@GiovaniFerraroTrivelli Lo dije en base a lo que Mike comento, "Salud" como tal puede no verse bien en resoluciones bajas, otro sinonimo cercano y acertivo seria "Vida" y capaz tampoco sirva, es una letra menos, ademas, creo que estamos buscando la traduccion al Español de "Health" por lo cual no creo que "HP" sea una opcion la verdad, yo optaria sinceramente por un cambio en como se muestra el Scoreboard, y si es necesario, ampliarlo para permitir cambios como estos y que no se sobrepongan los textos.

@mikela-valve Scoreboard will need a rework, I am not sure if there is a valid option to make every string size correctly into the Scoreboard without making a huge change on it. As you can see, certain strings on Spanish or Russian texts are not being displayed correctly and there are not very good words to replace them. I would like to know if you got something on mind to fix scoreboard problems with larger strings.

**Comment by afwn90cj93201nixr2e1re** [comment]
```
cstrike/cl_dlls/client.dll
hw.dll
ServerBrowser.dll
sw.dll
swds.dll
valve/cl_dlls/GameUI.dll
```
string: resource/icon_close
![image](https://user-images.githubusercontent.com/9306996/59071161-9373ef80-88ce-11e9-828b-3c31bd3ef3e1.png)
![image](https://user-images.githubusercontent.com/9306996/59071239-dfbf2f80-88ce-11e9-9171-c1c8a6801fa1.png)

**Comment by Maxi605** [comment]
> @GiovaniFerraroTrivelli Lo dije en base a lo que Mike comento, "Salud" como tal puede no verse bien en resoluciones bajas, otro sinonimo cercano y acertivo seria "Vida" y capaz tampoco sirva, es una letra menos, ademas, creo que estamos buscando la traduccion al Español de "Health" por lo cual no creo que "HP" sea una opcion la verdad, yo optaria sinceramente por un cambio en como se muestra el Scoreboard, y si es necesario, ampliarlo para permitir cambios como estos y que no se sobrepongan los textos.
> 
> @mikela-valve Scoreboard will need a rework, I am not sure if there is a valid option to make every string size correctly into the Scoreboard without making a huge change on it. As you can see, certain strings on Spanish or Russian texts are not being displayed correctly and there are not very good words to replace them. I would like to know if you got something on mind to fix scoreboard problems with larger strings.

También se podría cambiar el "Latencia" a "Ping" o "Lag", nunca veo a nadie decir "tengo 30 de latencia", mayormente dicen "tengo 30 de ping", si lo que se busca es ahorrar caracteres para que más cosas entren podría quedar bien.

**Comment by oaus** [comment]
> > @GiovaniFerraroTrivelli Lo dije en base a lo que Mike comento, "Salud" como tal puede no verse bien en resoluciones bajas, otro sinonimo cercano y acertivo seria "Vida" y capaz tampoco sirva, es una letra menos, ademas, creo que estamos buscando la traduccion al Español de "Health" por lo cual no creo que "HP" sea una opcion la verdad, yo optaria sinceramente por un cambio en como se muestra el Scoreboard, y si es necesario, ampliarlo para permitir cambios como estos y que no se sobrepongan los textos.
> > @mikela-valve Scoreboard will need a rework, I am not sure if there is a valid option to make every string size correctly into the Scoreboard without making a huge change on it. As you can see, certain strings on Spanish or Russian texts are not being displayed correctly and there are not very good words to replace them. I would like to know if you got something on mind to fix scoreboard problems with larger strings.
> 
> También se podría cambiar el "Latencia" a "Ping" o "Lag", nunca veo a nadie decir "tengo 30 de latencia", mayormente dicen "tengo 30 de ping", si lo que se busca es ahorrar caracteres para que más cosas entren podría quedar bien.

Sería mejor "Ping"

**Comment by d3m37r4** [comment]
It may be better to introduce short names for columns. Abbreviations...
Health = HP
Money = $
Score = K (kills)
Deaths = D
And Latency = Ping.

**Comment by Maxi605** [comment]
> It may be better to introduce short names for columns. Abbreviations...
> Health = HP
> Money = $
> Score = K (kills)
> Deaths = D
> And Latency = Ping.

Would be a bit confusing

**Comment by d3m37r4** [comment]
> Would be a bit confusing
>

![изображение](https://user-images.githubusercontent.com/26052038/59084516-bbdb0880-8925-11e9-82bc-2fed29c65dc7.png)
In csgo, it doesn't look confusing, another thing is to get used to it.

**Comment by RauliTop** [comment]
> It may be better to introduce short names for columns. Abbreviations...
> Health = HP
> Money = $
> Score = K (kills)
> Deaths = D
> And Latency = Ping.

That for all languages and we avoid many many MANY problems.......

I said it two weeks ago.

**Comment by basuritashka** [comment]
I mean, that will works if we add them on a "global" localization file that will be used regardless of our language, it is a good option, as I said, we will just wait for @mikela-valve, if we need to stick with actual specific strings to each language or just request a global localization for certain specific strings. (Like HP, $, K, D or Ping)

**Comment by afwn90cj93201nixr2e1re** [comment]
there no problems with locals, but align/margin problem's in tab. It would be nice if mikela can make some auto size column's for this.

**Comment by afwn90cj93201nixr2e1re** [comment]
In cs:go there no health column in tab:
https://github.com/ValveSoftware/halflife/issues/702#issuecomment-499769970
Maybe we should just add it in figures after nickname? And just truncate nickname like: 
```js
truncate(value, length) {return value.length > length ? value.substr(0, length) + '...': value;}
```
?
`Player (HP or Health: %d)`
or
`Player | Health: %d`
or
`Player12... | HP: %d`

**Comment by afwn90cj93201nixr2e1re** [comment]
`void __cdecl CCSClientScoreBoardDialog::UpdatePlayerAvatar(CCSClientScoreBoardDialog *const this, int playerIndex, KeyValues *kv)`
in this function we get steam players avatar's and we already have implemented `resource/icon_blocked` and `resource/icon_friend_indicator_scoreboard` simple way is replace text for BOMB/VIP/Dead/Defuse kit with icon's. It takes less space (for example 16x16 like close button in dialog's, which description you can find above) and there no internationalization problem's.

**Comment by d3m37r4** [comment]
> 
> 
> `void __cdecl CCSClientScoreBoardDialog::UpdatePlayerAvatar(CCSClientScoreBoardDialog *const this, int playerIndex, KeyValues *kv)`
> in this function we get steam players avatar's and we already have implemented `resource/icon_blocked` and `resource/icon_friend_indicator_scoreboard` simple way is replace text for BOMB/VIP/Dead/Defuse kit with icon's. It takes less space (for example 16x16 like close button in dialog's, which description you can find above) and there no internationalization problem's.

It would be perfect, we will wait for a response from @mikela-valve .

**Comment by RauliTop** [comment]
> In cs:go there no health column in tab:
> [#702 (comment)](https://github.com/ValveSoftware/halflife/issues/702#issuecomment-499769970)
> Maybe we should just add it in figures after nickname? And just truncate nickname like:
> 
> ```js
> truncate(value, length) {return value.length > length ? value.substr(0, length) + '...': value;}
> ```
> 
> ?
> `Player (HP or Health: %d)`
> or
> `Player | Health: %d`
> or
> `Player12... | HP: %d`

Bad idea. Players will change his name to HP: %d or whatever to confuse or disturb...

**Comment by PartialCloning** [comment]
If not all columns can fit, you could remove health and call it a day? For the most part gameplay wise it's not so useful. It's nice for mods but a new issue requesting a new column for mods seems more appropriate.

You could add an option in the Account message or a server sided cvar to display or not display the $ symbol. Making the money column a multipurpose column.

**Comment by Maxi605** [comment]
> If not all columns can fit, you could remove health and call it a day? For the most part gameplay wise it's not so useful. It's nice for mods but a new issue requesting a new column for mods seems more appropriate.
> 
> You could add an option in the Account message or a server sided cvar to display or not display the $ symbol. Making the money column a multipurpose column.

Yeah, i don't see much use for 'Health' on the scoreboard.

**Comment by mikela-valve** [comment]
> > It may be better to introduce short names for columns. Abbreviations...
> > Health = HP
> > Money = $
> > Score = K (kills)
> > Deaths = D
> > And Latency = Ping.
> 
> That for all languages and we avoid many many MANY problems.......
> 
> I said it two weeks ago.



> I mean, that will works if we add them on a "global" localization file that will be used regardless of our language, it is a good option, as I said, we will just wait for @mikela-valve, if we need to stick with actual specific strings to each language or just request a global localization for certain specific strings. (Like HP, $, K, D or Ping)

I can actually just do this in the logic for building the column headers rather than relying on the strings coming from the localization files. What I think should be done about this for now is that I should add another cvar, `scoreboard_shortenheaders` and if set it will force the headers to show as @RauliTop suggested and if not (by default) it will show the localized versions of the headers. Then in future releases (like when I work on showing avatars at lower resolutions) I'll revisit the column headers and figure out what else can be done to fit them better for different languages.

@GiovaniFerraroTrivelli I checked the Spanish localization in different resolutions and all of the strings seem to fit without changing `Salud` so it should be alright to leave that. The Score/Deaths/Latency headers are very tight at 640x480 but fit right against each other so I'll leave it how it is for now.

**Comment by afwn90cj93201nixr2e1re** [comment]
Why you can't just add icons?

**Comment by mikela-valve** [comment]
@afwn90cj93201nixr2e1re I'm actually not opposed to switching the class strings to icon instead but it's going to take a bit of extra work to add that. 

Some of the desirable icons like the defusal tool and bomb HUD icons are actually sprites and not images which aren't don't look to be usable easily in VGUI panels like the scoreboard, so those will have to be extracted and probably edited for transparency before they'll be usable.  Other icons either will also have to be edited, like if icon_skulls from CZ is used for death, or created, like a V or VIP icon. 

For now I'd like to leave things as they are with the `scoreboard_shortheaders` cvar and `scoreboard_show{health | money}` cvar's letting you control how things are laid out on the scoreboard and revisit other improvements like icons later. I'd appreciate it if you wanted to make another issue to track that as well as the icon fix that you mention [here](https://github.com/ValveSoftware/halflife/issues/702#issuecomment-499696424).

**Comment by mikela-valve** [comment]
`scoreboard_shortheaders` is added as of beta update '15:46:27 Jun 11 2019 (8265)'.

**Comment by afwn90cj93201nixr2e1re** [comment]
> I'm actually not opposed to switching the class strings to icon instead but it's going to take a bit of extra work to add that.

just copy paste it and add icon's to resourse dir.

Every thing already implemented in ScoreBoard classes in src.

**Comment by CS-PRO1** [comment]
If something like CS:S icons in scoreboard is possible it'll be great! ex [1](https://www.gamingcfg.com/img/16/scoreboard-css.jpg) [2](http://i62.servimg.com/u/f62/12/52/96/86/de_dus11.jpg)
I may be able to help with converting sprites to transparent images.

**Comment by afwn90cj93201nixr2e1re** [comment]
Yeah. Im' talking 'bout this or csgo icons.

**Comment by perforatorRU** [comment]
Add HLTV flag like VIP or BOMB

**Comment by GiovaniFerraroTrivelli** [comment]
> > I'm actually not opposed to switching the class strings to icon instead but it's going to take a bit of extra work to add that.
> 
> just copy paste it and add new icon's to resourse dir.
> Every thing already implemented in ScoreBoard classes in src.
>

This is not how things work. Personally, I like your suggestion but please be patient. There is a lot of work to do and adding things like that will delay times. We must be grateful for all of the work mikela is doing. If it were simple, he would have already done it 👍

**Comment by PartialCloning** [comment]
Looks good. Works good. Release it for all to enjoy.

**Comment by 10th** [comment]
> «15: 46: 27 июня 11 2019 (8265)»
Hello! After this update, 640х480, when connected (before spawn) or on older builds:

![de_dust20009](https://user-images.githubusercontent.com/11433503/59946071-233e9f80-9472-11e9-9ba4-2e4f64bd1e48.png)

**Comment by mikela-valve** [comment]
Thanks @10th, this should be fixed in the beta update I just made.

**Comment by 10th** [comment]
Thank you, @mikela-valve

A couple more thoughts:
1. Score (Kills + Rounds Won). It's not just K (Kills). Maybe K->S, Ping->P(L)?
2. All values can often be in the form of 1000 (mods and public servers). Therefore, the meaning of the reduction is not enough, because with these values in a shortened version, there are no spaces between the columns (640х480).
3. Ping it would be possible to reduce 1000->1k, but the rest still can't abbreviate.

![111](https://user-images.githubusercontent.com/11433503/59960072-eeab0200-94ca-11e9-856b-df263d7d6ae7.png)

**Comment by mikela-valve** [comment]
Thanks for the suggestions @10th. Changing K->S is a good idea, Ping I might leave as is to size the column and restrict the values displayed to 0-999, then 1.0k-99k.

When you say values for S/D will be in 1000’s, what range should I be expecting? Would 0-99999 be enough or does it need to show higher values than that?

**Comment by 10th** [comment]
I have not seen values above 9999.

**Comment by mikela-valve** [comment]
Great, I’ll just add a bit of space between the columns then so they don’t show so close when 1000 and above.

**Comment by afwn90cj93201nixr2e1re** [comment]
@mikela-valve don't do this. :/
That's so stupid to make abbreviations.
1253
after change gonna be 1.2k

I wanna see full integer, not rounded.

**Comment by mikela-valve** [comment]
Do you care as much about the tens and ones when the ping is >1000 though? I figured it might be nicer if it didn’t change as much. I can leave that as-is, though I’ll need to abbreviate everything >10000 at the moment to have it fit without making the column wider to accommodate it.

**Comment by afwn90cj93201nixr2e1re** [comment]
So, about wider. On old tab everything work's fine. Here no real cases in normal game(without custom mod's) where values in column's can be greater than 1000-9999.
Realy, no real cases.
Just make it simple.
Or rewrite tab fully. There no another way's. We can wait. Till August, there every year update's in August.

**Comment by afwn90cj93201nixr2e1re** [comment]
![изображение](https://user-images.githubusercontent.com/9306996/59960574-27eb6e00-94db-11e9-9de9-8b4f261ef02a.png)
Then there no problems with aligment.
Just test it for first time in 640x480, if it's good, than another resolution's work's fine too.


Or:
![изображение](https://user-images.githubusercontent.com/9306996/59960620-e8715180-94db-11e9-9d81-6bd4e296e94e.png)


Idk.
Old style table is better.

**Comment by 10th** [comment]
The background of the current player's row is rather dim. Is it possible to slightly reduce the transparency or add juiciness to the color?

![222](https://user-images.githubusercontent.com/11433503/59960743-6e3dce80-94d5-11e9-8c8e-19ed52003a8b.png)

**Comment by CS-PRO1** [comment]
I guess I'm doing the icons now.. I need to know 2 things: 
1. So what color they should be. 
2. They should have actual transparent background or a pure blue/pink/green background?
@mikela-valve

**Comment by mikela-valve** [comment]
The biggest issue to tackle I believe is making icons that will be quickly identifiable at the smaller line sizes used in <768 heights. Especially once you have >24 players on the server the line is quite tiny.

Probably the current HUD icons will be the most recognizable, so starting there would be best and keeping them similar to the green and transparency of the HUD will make them neutral in colour (i.e. not looking like a header or T/CT-specific) and will stand out in the row.

I'll have to check the line sizes to actually know what they are, I think they're proportionally set based on the font.

**Comment by mikela-valve** [comment]
Update 8279 has a few changes in it. 

@10th the header widths have been increased slightly so they shouldn't run so closely together. Also the player's highlight has been slightly intensified to make it more noticeable.

This update also enables avatars showing in <768 heights, so that dead space on the right of the scoreboard is finally used. I tested them at all of the different line sizes for different numbers of players but let me know if anything looks off.

**Comment by afwn90cj93201nixr2e1re** [comment]
> Especially once you have >24 players on the server the line is quite tiny.
avatar's same, but seems nice, Here no problems with size. 

Keep same size's.

**Comment by 10th** [comment]
@mikela-valve Cool, thank you!

**Comment by dager10** [comment]
my vision how scoretab need to look like with short abbreviation, $ - money, H - health, K - kills, D - deaths, L - latency (or P - ping). We also can show money without $ symbol.

![scoretab](https://user-images.githubusercontent.com/40521420/60180401-a07a6380-9828-11e9-8019-e43198a40866.png)

**Comment by afwn90cj93201nixr2e1re** [comment]
Not bad.
But on some server's hp can be more than 9999, and there is problem.
Coz on low resolution there no extra space for additional columns.

**Comment by dager10** [comment]
of course we can do less space between dead status and money and more space between money and health and i think 5 digits numbers will be display good enough for fields $, H, K, D, L, but need to test it

**Comment by CS-PRO1** [comment]
Would fonts work better instead of actual images? like CS:S for example..

**Comment by agrastiOs** [comment]
I think this is done honestly, all these are other feature suggestions that should be posted separately. What do you think @kisak-valve?

**Comment by afwn90cj93201nixr2e1re** [comment]
It's breakable change for low resolution's as you can see ( https://github.com/ValveSoftware/halflife/issues/702#issuecomment-499692877 ), we should provide more pleasant solution.
@agrastiOs

**Comment by Zetz512** [comment]
@afwn90cj93201nixr2e1re about the HP and amount, can add a cvar to change how you see the HP in value or in percentage like mp_healthscoreboard_mode

**Comment by afwn90cj93201nixr2e1re** [comment]
Now everything is fine, there many unnecessary cvar's from March update's.
We can get rid of them. If @mikela-valve gonna make some improvment's, xD.

**Comment by mikela-valve** [comment]
@afwn90cj93201nixr2e1re Which cvar's are unnecessary? Also, are you still seeing issues with the columns like in [#702 (comment)](https://github.com/ValveSoftware/halflife/issues/702#issuecomment-499692877)? The health and money columns should be stealing space from the name column, not pushing any of the other columns to the right.

**Comment by Splatt581** [comment]
1. In the scoreboard, the health field is referred to as `HP` for short, regardless of the value of the cvar `scoreboard_shortheaders`. I suggest giving this field the full name `Health`, if `scoreboard_shortheaders` is `0`.

![de_dust0004](https://user-images.githubusercontent.com/31417200/60684529-39efe800-9eaf-11e9-83f1-99b1b45024ff.jpg)
![de_dust0005](https://user-images.githubusercontent.com/31417200/60684534-41af8c80-9eaf-11e9-83ab-55c27419b2fc.jpg)

2. With `mp_scoreboard_showmoney` / `showhealth 0`, the server still sends `HealthInfo` and `Account` messages, but with empty fields. Is this normal server behavior? Sample packet below:

![newmsgs](https://user-images.githubusercontent.com/31417200/60684554-5f7cf180-9eaf-11e9-8535-12d6b9939e64.JPG)

3. When we first open the scoreboard, the avatars of the players slightly "jump" to the left. This happens at all resolutions:

![avatar](https://user-images.githubusercontent.com/31417200/60684567-7885a280-9eaf-11e9-9991-ec83f62c754f.gif)

**Comment by 10th** [comment]
@Splatt581, 
3. And this happens not only at the first opening, but also when new players enter, when their avatars are loaded.

**Comment by mikela-valve** [comment]
Thanks @Splatt581 and @10th. 

1. I had the column labeled `Health` originally but switched to `HP` as a standard header label for it to standardize it across localizations, that is it will just be `HP` in all languages. That's why it's now `HP` both with `scoreboard_shortheaders` set and unset.

2. Those messages are sent so that the clients know whether or not to show those fields, regardless of when the client connects rather than add or updates messages to set settings on the client. For simplicity the server just sends periodic updates at a low rate (every 5s or when values change) to prevent any weird edge case issues like stale values sticking around through a team change or a server cvar change or things like that. I did notice it was sending health/money updates when the value changed regardless of whether the real value was sent so I did change it to skip those updates and only send the periodic update in that case.

Point 3 should be fixed in the next release, that column was being sized improperly when the 
scoreboard is first created then sized properly once a player update happens.

**Comment by afwn90cj93201nixr2e1re** [comment]
> Which cvar's are unnecessary? 
Can you show new cvar list which include new cvar's from April for example?

**Comment by afwn90cj93201nixr2e1re** [comment]
@mikela-valve

**Comment by uniarch** [comment]
I think this needs to be closed. money bar is done and shows numbers correctly. you can format values above 99,999 as 100k,666k,1m,1b... but is pretty nice as it is, low resolutions are looking good.
I kindly ask mikela to do this release asap and begin to resolve https://github.com/ValveSoftware/halflife/issues/2154,https://github.com/ValveSoftware/halflife/issues/12
The hitbox issue is one if not the most important issue and the one the community waits for a very long time.
And the second will open much greater possibilities for mappers and modders.

Please, don't hesitate with current release and make our dreams happen :3

**Comment by agrastiOs** [comment]
I agree this can be closed.

**Comment by uniarch** [comment]
> 
> 
> I think this needs to be closed. money bar is done and shows numbers correctly. you can format values above 99,999 as 100k,666k,1m,1b... but is pretty nice as it is, low resolutions are looking good.
> I kindly ask mikela to do this release asap and begin to resolve #2154 , #12
> The hitbox issue is one if not the most important issue and the one the community waits for a very long time.
> And the second will open much greater possibilities for mappers and modders.
> 
> Please, don't hesitate with current release and make our dreams happen :3

or if not closed then transferred to future release as too large :?

**Comment by CS-PRO1** [comment]
Yea I agree on releasing this as it is and having a new issue track for future/ongoing release.

**Comment by mikela-valve** [comment]
Closing this as fixed. We should create a new issue to track additional work on the interface like the addition of icons.

**Comment by PartialCloning** [comment]
@mikela-valve when will we see this update officially? When is "Next Release"?
